plugins/org.eclipse.fordiac.ide.model.commands/src/org/eclipse/fordiac/ide/model/commands/change/AbstractUpdateBlockFBNElementCommand.java

```diff
@@ -395,12 +395,12 @@ protected void handleParameters() {
 
 	private void processEvents(final InterfaceList interfaceList) {
 		for (final Event input : interfaceList.getEventInputs()) {
-			if (input.getInputConnections().isEmpty() || (!input.getAttributes().isEmpty())) {
+			if (!input.getInputConnections().isEmpty() || !input.getAttributes().isEmpty()) {
```

Comment:
As I understand it, this is supposed to specifically handle event pins *without* connections but *with* attributes. I believe it should be changed to the following instead[^1]:
```suggestion
			if (input.getInputConnections().isEmpty() && !input.getAttributes().isEmpty()) {
```

[^1]: see also the similar condition in `processVars(..)` below

Reply:
Good Point. I missed that we have a dedicated section checking for connections. When we finde some time this class would really need some love for beeing cleaned and duplicated loops to be harmonized. but that is not for a bug fix on 3.0.x.